localstack/services/apigateway/integration.py

```diff
@@ -276,11 +276,20 @@ def construct_invocation_event(
         single_value_query_string_params = {
             k: v[-1] if isinstance(v, list) else v for k, v in query_string_params.items()
         }
+        # Some headers get capitalized like in CloudFront, see
+        # https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/add-origin-custom-headers.html#add-origin-custom-headers-forward-authorization
+        # It seems AWS_PROXY lambda integrations are behind cloudfront, as seen by the returned headers in AWS
```

Comment:
nit: Out of curiosity - presumably that's only relevant for `AWS_PROXY` integrations, not for regular `AWS` integrations, right? Not something we need to cover here, was just curious if you had some insights into that.

Reply:
You're right! I had the same question, it should not be relevant for other `AWS` integrations because we selectively pass headers to the integration via request parameters.
However, I wonder if it's relevant to `HTTP_PROXY` integrations. However, we do not have good validated tests in community yet. But I keep a note for the next refactor for this kind of thing 😄 good to keep in mind! 👌

Reply:
@cloutierMat had a very good insight about this: this is only the case for Edge optimized API endpoints: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-endpoint-types.html

We should probably fix this in a next iteration, as this is quite the edge case, but this really good to know.